### ISSUE TITLE
Weechat crashes on launch under Ubuntu

### ISSUE THREAD

**Issue by bdesham** [NixOS/nixpkgs#64167]
## Issue description

When I run a freshly-installed Weechat, the app quits and I get a Python traceback:

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/site.py", line 563, in <module>
    main()
  File "/usr/lib/python2.7/site.py", line 545, in main
    known_paths = addusersitepackages(known_paths)
  File "/usr/lib/python2.7/site.py", line 272, in addusersitepackages
    user_site = getusersitepackages()
  File "/usr/lib/python2.7/site.py", line 247, in getusersitepackages
    user_base = getuserbase() # this will also set USER_BASE
  File "/usr/lib/python2.7/site.py", line 237, in getuserbase
    USER_BASE = get_config_var('userbase')
  File "/usr/lib/python2.7/sysconfig.py", line 582, in get_config_var
    return get_config_vars().get(name)
  File "/usr/lib/python2.7/sysconfig.py", line 528, in get_config_vars
    _init_posix(_CONFIG_VARS)
  File "/usr/lib/python2.7/sysconfig.py", line 412, in _init_posix
    from _sysconfigdata import build_time_vars
  File "/usr/lib/python2.7/_sysconfigdata.py", line 6, in <module>
    from _sysconfigdata_nd import *
ImportError: No module named _sysconfigdata_nd
```

I had no `~/.weechat` directory when I ran the command, although it created one for me before it quit.

My terminal is all messed up after I run this—my scrollback is cleared, and that traceback and all future commands’ outputs are printed as if `\n` moved to the next line but did not reset the cursor to the leftmost column. I assume that the app just set up some TUI stuff that it didn’t get the opportunity to undo.

As expected, `which weechat` says `/home/bdesham/.nix-profile/bin/weechat`.

### Steps to reproduce

1. `nix-env -iA nixpkgs.weechat`
2. `weechat`

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.4.0-151-generic, Ubuntu, 16.04.6 LTS (Xenial Xerus)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(bdesham): `"nixpkgs-19.09pre184398.d98a165de78"`
 - nixpkgs: `/home/bdesham/.nix-defexpr/channels/nixpkgs`

If I run the same steps on my Mac, I get what seems like a perfectly fine Weechat installation. Details for that machine:

 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.6.0, macOS 10.14.5`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.2.2`
 - channels(besham): `"nixpkgs-19.09pre184398.d98a165de78"`
 - nixpkgs: `/Users/besham/.nix-defexpr/channels/nixpkgs`

**Comment by bdesham** [comment]
Cutting Python out of the equation allowed Weechat to run. I added this to my config.nix:

```
myWeechat = pkgs.weechat.override {
  configure = { availablePlugins, ... }: {
    plugins = with availablePlugins; [ lua ];
  };
};
```

Then, running

1. `nix-env -iA nixpkgs.myWeechat`
2. `weechat`

worked fine.

**Comment by FRidh** [comment]
This is an impurity. Apparently it picks up your systems' Python which it should not do.

**Comment by bdesham** [comment]
Fixed by the merge of #64262.
